From a1424c6b36dfb9751a2ba7cfa2c5c581e6039da6 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 23 Oct 2014 17:11:50 -0400
Subject: [scsi] qla2xxx: Disable PCI device in shutdown handler

Message-id: <1414084311-36200-2-git-send-email-cdupuis@redhat.com>
Patchwork-id: 98325
O-Subject: [PATCH RHEL 7.1 e-stor 1/2] qla2xxx: Disable PCI device in shutdown handler.
Bugzilla: 1067179
RH-Acked-by: Ewan Milne <emilne@redhat.com>
RH-Acked-by: Rob Evers <revers@redhat.com>

From: Chad Dupuis <chad.dupuis@qlogic.com>

scsi commit id 61d41f610debd557b1c451338fa0afaac9dd8719

Disable the PCI device during shutdown to prevent any races with
other PCI code such as the AER handling code.

Signed-off-by: Chad Dupuis <chad.dupuis@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_os.c b/drivers/scsi/qla2xxx/qla_os.c
index 431f81a..92bba6e 100644
--- a/drivers/scsi/qla2xxx/qla_os.c
+++ b/drivers/scsi/qla2xxx/qla_os.c
@@ -2962,6 +2962,9 @@ qla2x00_shutdown(struct pci_dev *pdev)
  qla2x00_free_irqs(vha);
 
  qla2x00_free_fw_dump(ha);
+
+ pci_disable_pcie_error_reporting(pdev);
+ pci_disable_device(pdev);
 }
 
 static void
-- 
1.7.1